### ISSUE TITLE
[FEATURE] Support SQL warehouses for CRUD operations on databricks_sql_table

### ISSUE THREAD

**Issue by zcking** [databricks/terraform-provider-databricks#2372]
### Use-cases

CRUD operations to a `databricks_sql_table` require a Compute cluster (the `cluster_id` argument). The majority of the time I feel SQL operations like this are done from SQL Warehouses, not all purpose clusters. As an added benefit, with the addition of SQL Serverless, this would greatly speed up execution for creating tables from a "cold start".

This feature request is to be able to use the `databricks_sql_table` resource with SQL Warehouses instead of Compute Clusters. 

### Attempted Solutions

I initially attempted to use a SQL warehouse ID in the `cluster_id` argument for the `databricks_sql_table` resource:

```terraform
resource "databricks_sql_table" "this" {
  catalog_name       = databricks_schema.this.catalog_name
  schema_name        = databricks_schema.this.name
  name               = "mytable"
  data_source_format = "PARQUET"
  table_type         = "EXTERNAL"
  storage_location   = var.storage_location
  cluster_id         = var.sql_warehouse_id # only works with Compute Cluster IDs, not SQL warehouses
  properties = {
    "tags.DeployedBy" = "Terraform"
  }
}
```

Provider version: 1.18.0
```
provider[registry.terraform.io/databricks/databricks] >= 1.18.0
```

### Proposal

Because there could be ambiguity differentiating Compute Cluster IDs from SQL Warehouse IDs, I'd like to propose a new optional string argument `warehouse_id` which should be mutually exclusive from `cluster_id`. And therefore, also should have backwards compatibility with the existing provider. 

For example, the usage above would become:

```terraform
resource "databricks_sql_table" "this" {
  catalog_name       = databricks_schema.this.catalog_name
  schema_name        = databricks_schema.this.name
  name               = "mytable"
  data_source_format = "PARQUET"
  table_type         = "EXTERNAL"
  storage_location   = var.storage_location
  # cluster_id = "" <-- cannot use this argument if using the `warehouse_id` argument
  warehouse_id         = var.sql_warehouse_id
  properties = {
    "tags.DeployedBy" = "Terraform"
  }
}
```

**Comment by nfx** [comment]
Warehouse and cluster REST query execution APIs are incompatible.

**Comment by zcking** [comment]
@nfx they are incompatible with what? Each other? Can you elaborate, is your response that it can't be done or that it's too much effort to add and maintain in the provider? 

The SQL execution API appears completely compatible with this use case to me, but maybe I'm overlooking something:
https://docs.databricks.com/api/azure/workspace/statementexecution/executestatement

Thanks

**Comment by nkvuong** [comment]
@zcking 
This SQL execution API is only supported by the SQL warehouse - https://docs.databricks.com/api/azure/workspace/statementexecution/executestatement

The current implementation is done using this API on an all-purpose cluster - https://docs.databricks.com/api/azure/workspace/commandexecution/execute. This is simply because SQL Exec API was not widely available back then

Supporting SQL warehouse is a sensible request, we will add it to the backlog

**Comment by zcking** [comment]
@nkvuong that's understandable, even now it is in Public Preview and not GA. 

I would be open to contributing on this as well if needed. I haven't contributed to this project before but I have a lot of Go experience and some familiarity with Terraform provider implementations.

**Comment by zcking** [comment]
From what I can see, the `commands` package encapsulates all the API requests to the existing command execution API on all purpose clusters. 

I can think of a couple of options to implement the execution API calls that will be necessary for this:

Option 1 is we could retrofit the `commands` package to additional support executing on SQL warehouses/endpoints

Option 2 is build a smaller unit function in the `sql` package that encapsulates the SQL statement execution API alone. 

I feel that Option 2 is cleaner and easier to maintain, but would appreciate any feedback from the maintainers if there's a strong preference or other ideas. Thanks

**Comment by alexott** [comment]
My personal choice would be option 2 - `commands` isn't limited to SQL commands only.

**Comment by zcking** [comment]
I have a WIP that I pushed. I created a wrapper around the SQL Statement Execution API at `sql/statement_execution.go` but based it on the `commands` package and only implemented the bare minimum for this feature. 

I wrote some initial unit tests for this, as well as my changes to the actual resource `databricks_sql_table`. 

Any feedback about this work in progress would be greatly appreciated as I'm new to the codebase. Thanks!
